metadata/src/main/java/org/apache/kafka/controller/ControllerPurgatory.java

```diff
@@ -65,6 +81,7 @@ void failAll(Exception exception) {
         while (iter.hasNext()) {
             Entry<Long, List<DeferredEvent>> entry = iter.next();
             for (DeferredEvent event : entry.getValue()) {
+                log.info("failAll({}): failing {}.", exception.getClass().getSimpleName(), event);
```

Comment:
I see. We don't log the stacktrace because `failAll` is only called with `NotControllerException`. Should we change the signature of this method to `void failAll(ApiException)`?

Reply:
The rationale for not giving the stacktrace was more brevity, than the fact that it was an `ApiException`. So maybe we can leave the signature as-is?